main/src/main/java/org/mobilitydata/gtfsvalidator/validator/TripAndShapeDistanceValidator.java

```diff
@@ -32,55 +35,75 @@
+              int nbStopTimes = stopTimeTable.byTripId(trip.tripId()).size();
+              if (nbStopTimes == 0) {
+                return;
+              }
+              GtfsStopTime lastStopTime =
+                  stopTimeTable.byTripId(trip.tripId()).get(nbStopTimes - 1);
+              GtfsStop stop = stopTable.byStopId(lastStopTime.stopId()).orElse(null);
+              if (stop == null) {
+                return;
+              }
+              double maxStopTimeDist = lastStopTime.shapeDistTraveled();
+
+              // Get max shape distance for trip
+              GtfsShape maxShape =
                   shapeTable.byShapeId(shapeId).stream()
-                      .mapToDouble(GtfsShape::shapeDistTraveled)
-                      .max()
-                      .orElse(Double.NEGATIVE_INFINITY);
-
-              tripTable
-                  .byShapeId(shapeId)
-                  .forEach(
-                      trip -> {
-                        double maxStopTimeDist =
-                            stopTimeTable.byTripId(trip.tripId()).stream()
-                                .mapToDouble(GtfsStopTime::shapeDistTraveled)
-                                .max()
-                                .orElse(Double.NEGATIVE_INFINITY);
-
-                        if (maxStopTimeDist > maxShapeDist) {
-                          noticeContainer.addValidationNotice(
-                              new TripDistanceExceedsShapeDistanceNotice(
-                                  trip.tripId(), shapeId, maxStopTimeDist, maxShapeDist));
-                        }
-                      });
+                      .max(Comparator.comparingDouble(GtfsShape::shapeDistTraveled))
+                      .orElse(null);
+              if (maxShape == null) {
+                return;
+              }
+
+              double maxShapeDist = maxShape.shapeDistTraveled();
+              double distanceInMeters =
+                  getDistanceMeters(maxShape.shapePtLatLon(), stop.stopLatLon());
+              if (maxStopTimeDist > maxShapeDist) {
+                if (distanceInMeters > DISTANCE_THRESHOLD) {
+                  noticeContainer.addValidationNotice(
+                      new TripDistanceExceedsShapeDistanceNotice(
+                          trip.tripId(), shapeId, maxStopTimeDist, maxShapeDist, distanceInMeters));
+                } else if (distanceInMeters > 0) {
+                  noticeContainer.addValidationNotice(
+                      new TripDistanceExceedsShapeDistanceBellowThresholdNotice(
+                          trip.tripId(), shapeId, maxStopTimeDist, maxShapeDist, distanceInMeters));
+                }
+              }
             });
   }
 
-  /** The distance traveled by a trip should be less or equal to the max length of its shape. */
+  /**
+   * The distance traveled by a trip should be less or equal to the max length of its shape.
+   *
+   * <p>The distance is greater or equal to the 11.1m threshold.
```

Comment:
Revised to clarify threshold: `The distance between the last shape point and last stop point is greater than or equal to the 11.1m threshold.`